monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/NotificationManager.java

```diff
@@ -65,6 +66,31 @@ public boolean addRecipient(String name, String email, List<String> appIds) thro
         }
         return true;
     }
+    
+    @Override
+    public boolean modifyRecipient(String email, String name, List<String> apps) throws NoEmailRecipientException {
+        // Code here
+        if (email == null || email.trim().isEmpty()) {
+            return false;
+        }
+        if (dbConnection.getById(email) == null) {
+            throw new NoEmailRecipientException(); // Throw a exception instead
+        }
+        INotificationRecipient recipient = this.getRecipient(email);
+        recipient.setName(name);
+        recipient.setEmail(email);
+        recipient.setApps(new ArrayList<>());
+        for (String id : apps) {
+            recipient.getApps().add((App) appManager.getApp(id));
+        }
+        try {
+            dbConnection.update(recipient);
+        } catch (UnstorableObjectException e) {
+            // should not happen
+            logger.error("Could not store recipient.", e);
+        }
+        return true;
+    }
```

Comment:
My reason for not merging these two methods is that the method addRecipient() checks if the email already exists and throws a EmailAlreadyRegisteredException and the method modifyRecipient() should not throw this as the email should always exist. If I had to add the modify functionality in addRecipient(), I would have to change the method signature to not check if the email already exists. But this would trump the necessary check at the service layer and throw a database exception instead, which didn't feel like a proper way of handling the error. Could you please advise if you still want me to merge these methods and how I should go about it?

Reply:
Ya, I see the slight differences. You could at least extract the code to set the recipient values though (line 80-85)